src/main/java/com/sapient/controller/ExpenseController.java

```diff
@@ -29,12 +33,51 @@ public Record expense(@Argument String passwordHash, @Argument Integer id) {
     	}
     }
 
+    @QueryMapping
+    public Record expenses(@Argument String passwordHash) {
+        try {
+            List<Expense> found = expenseService.getExpenses(passwordHash);
+            return new ExpensesSuccess(found);
+        } catch (Exception e) {
+            return new FailurePayload(e.getClass().getSimpleName(), e.getMessage());
+        }
+    }
+
     // Properties of a returned object from a @SchemaMapping method map to graphql fields of the same name.
     // Eg: "exceptionName".
     @MutationMapping
-    public Record createExpense(@Argument String passwordHash, @Argument String title, @Argument String description, @Argument Double amount) {
+    public Record createExpense(@Argument String passwordHash, @Argument String title, @Argument String description,
+                                @Argument Double amount, @Argument Integer epochDate, @Argument Integer categoryId,
+                                @Argument Integer merchantId, @Argument Integer recurrenceId) {
+
+        Date date;
+        if(epochDate == null){
```

Comment:
Since we are changing epochDate to be mandatory in the graphql, this check is unnecessary because it will never be null.